### PR TITLE
Sync upstream changes to 2.0

### PR DESCRIPTION
**Pull request by dhruvilshah3**

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java [hunk 1]
```diff
@@ -1156,6 +1157,80 @@ public void testResetOfProducerStateShouldAllowQueuedBatchesToDrain() throws Exc
...
+        final long producerId = 343434L;
+        TransactionManager transactionManager = new TransactionManager();
+        transactionManager.setProducerIdAndEpoch(new ProducerIdAndEpoch(producerId, (short) 0));
+        setupWithTransactionState(transactionManager);
+
+        Metrics m = new Metrics();
+        SenderMetricsRegistry senderMetrics = new SenderMetricsRegistry(m);
+
+        Sender sender = new Sender(logContext, client, metadata, this.accumulator, true, MAX_REQUEST_SIZE, ACKS_ALL, 10,
+            senderMetrics, time, REQUEST_TIMEOUT, 50, transactionManager, apiVersions, BATCH_EXPIRY_TIMEOUT);
```
Comment (dhruvilshah3) [inline clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java]: Added `BATCH_EXPIRY_TIMEOUT`.

File: clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java [hunk 2]
```diff
@@ -1156,6 +1157,80 @@ public void testResetOfProducerStateShouldAllowQueuedBatchesToDrain() throws Exc
...
+    public void testForceCloseWithProducerIdReset() throws Exception {
+        TransactionManager transactionManager = new TransactionManager();
+        transactionManager.setProducerIdAndEpoch(new ProducerIdAndEpoch(1L, (short) 0));
+        setupWithTransactionState(transactionManager);
+
+        Metrics m = new Metrics();
+        SenderMetricsRegistry senderMetrics = new SenderMetricsRegistry(m);
+
+        Sender sender = new Sender(logContext, client, metadata, this.accumulator, true, MAX_REQUEST_SIZE, ACKS_ALL, 10,
+            senderMetrics, time, REQUEST_TIMEOUT, 50, transactionManager, apiVersions, BATCH_EXPIRY_TIMEOUT);
```
Comment (dhruvilshah3) [inline clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java]: Added `BATCH_EXPIRY_TIMEOUT`.

**Comment by cmccabe** [comment]
LGTM

**Comment by dhruvilshah3** [comment]
Thanks, merged.

**Pull request merged**